### PR TITLE
Add a registry-based system for plugins

### PR DESCRIPTION
**Pull request by claudep**
This is a follow-up to the 07dac39d3, where we try to delegate much of the code specific to particular plugins to the plugins code. There might be some more efforts to do to achieve this final goal, but I think this is already a good step forward.

### CODE REVIEW

**Comment by claudep** [comment]
@moggers87 I would love to get your input on this. I'll gladly fix the code coverage part if the general idea is approved.

**Review by moggers87 — COMMENTED** [review]
This looks really promising :+1:

**Comment by moggers87** [comment]
I have requested a review from @moggers87

**Comment by claudep** [comment]
Oh, that's a good friend of yours? :-)

**Comment by claudep** [comment]
@moggers87 I can totally understand that you can't find the time to review this, or more generally to make this package evolve.
Just tell me, and I'll try to find other reviewers.

File: two_factor/plugins/phonenumber/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import django
+
+if django.VERSION <= (3, 2):
```
Comment (moggers87) [inline two_factor/plugins/phonenumber/__init__.py:3]: Is this line (and the one for yubikey) necessary? I.e. will Django 4.0 start spitting out warnings if we don't put in this if statement

**Review by moggers87 — APPROVED** [review]
One question, but but approved either way.

This is a great improvement and certainly makes the prospect of adding new methods less daunting (from a maintenance point of view at least)

**Comment by moggers87** [comment]
> @moggers87 I can totally understand that you can't find the time to review this, or more generally to make this package evolve. Just tell me, and I'll try to find other reviewers.

Sorry about the wait, I wanted to find the time/energy to give this a proper review. This is some good work :smile_cat:

**Review by moggers87 — CHANGES_REQUESTED** [review]
Ha! Spoke too soon: `two_factor.plugins.yubikey.utils` has zero test coverage and I think you just forgot to remove that file.

File: two_factor/plugins/phonenumber/__init__.py (hunk 1 above)
Comment (claudep) [reply two_factor/plugins/phonenumber/__init__.py:3]: Yes, it is. See 1b6b6a3f9 for the equivalent commit for the main `two_factor` app.
See https://github.com/django/django/blob/stable/4.0.x/django/apps/config.py#L171 for the location where Django raises the warning.

**Comment by claudep** [comment]
Ah yes, good catch! I removed that file now.

**Review by moggers87 — APPROVED** [review]

**Pull request merged**